coral/src/app/features/browse-topics/BrowseTopics.tsx

```diff
@@ -2,35 +2,32 @@ import { useGetTopics } from "src/app/features/browse-topics/hooks/topic-list/us
 import { Pagination } from "src/app/components/Pagination";
 import SelectTeam from "src/app/features/browse-topics/components/select-team/SelectTeam";
 import TopicList from "src/app/features/browse-topics/components/topic-list/TopicList";
-import { useGetTeams } from "src/app/features/browse-topics/hooks/teams/useGetTeams";
 import { useState } from "react";
 import { Flexbox, FlexboxItem } from "@aivenio/design-system";
 import { useSearchParams } from "react-router-dom";
 import { Environment } from "src/domain/environment";
 import SelectEnvironment from "src/app/features/browse-topics/components/select-environment/SelectEnvironment";
 import { useGetEnvironments } from "src/app/features/browse-topics/hooks/environment/useGetEnvironments";
 import { SearchTopics } from "src/app/features/browse-topics/components/search/SearchTopics";
+import { Team } from "src/domain/team";
 
-// Use a UUID value to represent empty option value.
-const ALL_TEAMS_VALUE = "f5ed03b4-c0da-4b18-a534-c7e9a13d1342";
 const ALL_ENVIRONMENTS_VALUE = "ALL";
 
 function BrowseTopics() {
   const [searchParams, setSearchParams] = useSearchParams();
+  const [teamName, setTeamName] = useState<Team | null>(null);
```

Comment:
I never use `undefined` as an explicit value (if I can avoid it), since debugging (incl. getting the right errors) is easier with `null`. `undefined` can happen for various reasons, but a `null` always tells you right away that it was set by the code on purpose.

Reply:
I have used the following as my justification:
> TypeScript team doesn't use null : [TypeScript coding guidelines](https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined) and it hasn't caused any problems. Douglas Crockford thinks [null is a bad idea](https://www.youtube.com/watch?v=PSGEjv3Tqo0&feature=youtu.be&t=9m21s) and we should all just use undefined.

https://basarat.gitbook.io/typescript/recap/null-undefined#final-thoughts

Reply:
https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined

Reply:
I mean, per definition `undefined` means that a variable is not (yet) assigned to a value, so it's implicitly not having no value. Undefined can happen by accident. This is a different kind of communication than a variable getting explicitly no value, which shows intent. For me, that improves readability. 

TS coding guideline has no argument why to use undefined over null and the video also has no real argument other than that's a built-in mistake (like so many 😅 ) by JS to have two values indicating no value.

I've always followed the different argument than the one Douglas Crockfords has - always leave `undefined` to the compiler, use `null` if you want to have a "no value" explicitly.  e.g. like described here: https://2ality.com/2021/01/undefined-null-revisited.html#my-approach

I mean, for me it's probably a leftover learning from before ES6 times 😅  so there's probably pro and con for whatever should be used to indicate a non-value. I read an `undefined` as something that was kind of out of control for the person writing the code, so for me `null` has more readability. But that's of course also based on familiarity with it, because I always worked in code bases following that 🤔

Reply:
> Conceptually, undefined indicates the absence of a value, while null indicates the absence of an object

- [MDN Reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#undefined_type)

I acknowledge that I am building my own bubble with arguments that fit into my views. I wonder how we will solve this tie. By using `useState` with the default value, it is also simpler 😄 .

Reply:
I mean, I also can build my own bubble 😅  I think most code-arguments are like that. We've TS and as long as we're consistent, I don't think there's a winner in `undefined` vs `null` tbh. We could also wait with deciding and bring the new addition to the FE team into that discussion. 

...which would be kind of mean at the first day tho 😂 

I stand by my point that readability/debuggability can profit from using `null`, but I also don't think it's a super important thing with our TS setup. So I would be fine doing with `undefined` and see how it goes.

Reply:
Ohhhhhhhhhhhhh maybe I've an idea how to solve this without having to use `undefined` or `null` 💡  I'll try tomorrow and report!